src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  addComment,
+  buttonAddLoading,
+  selectedPost,
+}) => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [bodyComment, setBodyComment] = useState('');
+  const [bodyCommentError, setBodyCommentError] = useState(false);
+
+  let newComment: CommentData;
```

Comment:
It's not recommended to declare a variable in the component scope and then mutate it in a function. It's better to create a new variable inside the function where you need it. This helps to avoid bugs related to variable hoisting and scope.